cloudnet/src/main/java/de/dytanic/cloudnet/template/install/installer/ServiceVersionInstaller.java

```diff
@@ -0,0 +1,27 @@
+package de.dytanic.cloudnet.template.install.installer;
+
+import de.dytanic.cloudnet.template.install.ServiceVersion;
+import de.dytanic.cloudnet.template.install.stream.ErrorConsoleStreamThread;
+import de.dytanic.cloudnet.template.install.stream.InfoConsoleStreamThread;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.nio.file.Path;
+import java.util.concurrent.CountDownLatch;
+
+public abstract class ServiceVersionInstaller {
+
+    public abstract void install(ServiceVersion version, Path workingDirectory, OutputStream targetStream) throws IOException;
+
+    protected int startProcessAndWaitFor(String command, Path workingDirectory) throws IOException, InterruptedException {
+        Process process = Runtime.getRuntime().exec(command, null, workingDirectory.toFile());
+        CountDownLatch countDownLatch = new CountDownLatch(2);
+        new ErrorConsoleStreamThread(process.getErrorStream(), countDownLatch).start();
+        new InfoConsoleStreamThread(process.getInputStream(), countDownLatch).start();
```

Comment:
Do you think it's important to show the output? And why no ExecutorService?

Reply:
Yeah, it is important

Reply:
_no_

Reply:
It is

Reply:
What if there are any errors? How would you fix them without having the console?

Reply:
Yeah ok, why the countdown latch?

Reply:
To wait until both streams are closed